### PR TITLE
drop use of deprecated pkg_resources

### PR DESCRIPTION
**Pull request by graingert**
re: #507

### CODE REVIEW

**Comment by graingert** [comment]
ah annoyingly this fails on python 3.7 due to the missing importlib.metadata vendoring importlib_metadata is a bit of a pain as it needs zipp and typing_extensions

**Comment by graingert** [comment]
actually I can do this with `setuptools.__version__` instead of `importlib.metadata.distribution("setuptools").version` because setuptools is already imported in `bdist_wheel.py`

**Comment by graingert** [comment]
Yeah I manually copied across the files imported by requirements.py https://github.com/pypa/wheel/pull/508/files#diff-3f63151da655b90d5d7ec0932887c5a073679a0917dfe94a9617a4e2806ae9caR8

**Review by agronholm — APPROVED** [review]
Looks good.

**Comment by agronholm** [comment]
Oh, one more thing: could you add a line to the changelog? I will merge then.

**Comment by graingert** [comment]
@agronholm done

File: docs/news.rst [hunk 1]
```diff
@@ -5,6 +5,7 @@ Release Notes
 
 - Updated vendored ``packaging`` to 23.0
 - Fixed spaces in platform names not being converted to underscores (PR by David Tucker)
+- Drop use of deprecated ``pkg_resources`` (PR by Thomas Grainger)
```
Comment (agronholm) [inline docs/news.rst:8]:
```suggestion
- Fixed ``DeprecationWarning`` about the use of the deprecated ``pkg_resources`` API
  (PR by Thomas Grainger)
```

**Comment by agronholm** [comment]
Thanks!

**Pull request merged**